sw/device/tests/kmac_entropy_test.c

```diff
@@ -0,0 +1,281 @@
+  LOG_INFO("entropy_period = %08X", entropy_period);
+}
+
+bool test_main(void) {
+  LOG_INFO("Running KMAC ENTROPY test...");
+
+  // Initialize KMAC HWIP.
+  dif_kmac_t kmac;
+  dif_kmac_operation_state_t kmac_operation_state;
+  dif_kmac_error_t err_reg;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC to use EDN entropy
+  // Set max values for the EDN timeout
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeEdn,
+      .entropy_fast_process = false,
+      .entropy_hash_threshold = 0x0,
+      .entropy_wait_timer = 0x0020,
+      .entropy_prescaler = 0x0001,
+      .message_big_endian = false,
+      .output_big_endian = false,
+      .sideload = false,
+      .msg_mask = false,
+  };
+
+  bool is_kmac_locked;
+  bool expected_timeout_status;
+  bool observed_timeout_status;
+  kmac_edn_timeout_t timeout_test;
+  size_t i = 0;
+
+  // Handle string encoding
+  dif_kmac_customization_string_t str_buffer;
+  CHECK_DIF_OK(dif_kmac_customization_string_init(
+      kmac_test_vector.customization_string,
+      kmac_test_vector.customization_string_len, &str_buffer));
+
+  // When customization_string is empty, use NULL to activate empty str path
+  dif_kmac_customization_string_t *s_buff =
+      kmac_test_vector.customization_string_len == 0 ? NULL : &str_buffer;
+
+  size_t digest_l =
+      kmac_test_vector.digest_len_is_fixed ? kmac_test_vector.digest_len : 0;
+  uint32_t out[DIGEST_LEN_KMAC_MAX];
+
+  for (; i < nr_of_tests; i++) {
+    LOG_INFO("loop_ctr = %d", i);
+    timeout_test = test_timeout_vals[i];
+
+    config.entropy_prescaler = timeout_test.prescaler;
+    config.entropy_wait_timer = timeout_test.wait_timer;
+    expected_timeout_status = is_timeout_expected(timeout_test);
+
+    CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+    CHECK_DIF_OK(dif_kmac_mode_kmac_start(&kmac, &kmac_operation_state,
+                                          kmac_test_vector.mode, digest_l,
+                                          &kmac_test_vector.key, s_buff));
+
+    // Absorbing stage
+    CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                                 kmac_test_vector.message,
+                                 kmac_test_vector.message_len, NULL));
+
+    CHECK(DIGEST_LEN_KMAC_MAX >= kmac_test_vector.digest_len);
+
+    // This is where timeout might happen, so we handle dif return manually
+    dif_result_t res = dif_kmac_squeeze(&kmac, &kmac_operation_state, out,
+                                        kmac_test_vector.digest_len, NULL);
+
+    // It is OK to get kDifError at this point because of possible timeout
+    CHECK(res == kDifOk || res == kDifError);
+    CHECK_DIF_OK(is_edn_timeout(&kmac, res, &observed_timeout_status));
+
+    if (observed_timeout_status) {
+      LOG_INFO("============================= EDN timed out.");
```

Comment:
nit: In DV style, we prefer not to use extra delineators like `======`. We don't see we use this kind of comment in C code either.

Reply:
I was using this to increase visibility in the printed messages, but I removed it for consistency.